### DISCUSSION TITLE
Get unit testing back

### DISCUSSION THREAD

**Discussion by mkruckow** [POSYDON-code/POSYDON#722] [Ideas]
We are saying for a long time, that we'd like to get the unit testing back.

Here we may want to have a list of stuff to be done for this (I'll start):

- there is no test grid: posydon/tests/data/POSYDON-UNIT-TESTS/visualization/grid_unit_test_plot.h5
This causes a `ValueError: Test grid for unit testing was not found!` when trying to do either of the following imports:
```
import posydon.tests.binary_evol.test_BinaryStar
import posydon.tests.binary_evol.test_SingleStar
import posydon.tests.visualization.test_plot1D
import posydon.tests.visualization.test_plot2D
import posydon.tests.binary_evol.SN.test_profile_collapse
```

**Comment by ssbvr** [comment]
I also suggest enforcing a code formatting convention. Everyone can adopt this if they come in a `.vs_code` file. I think it's a good idea to trigger the formation when saving a file. When this is in place and POSYDON has been reformatted, add stylistic automatic checks with a GitHub action so that future code will comply with the coding convention.

**Comment by mkruckow** [comment]
Here a progress list:

- [x] code files in the base `posydon`: `config.py`
- [ ] [in progress] code files in `bin` -> move code to utils
- [ ] code files in `active_learning`
- [ ] code files in `binary_evol`
- [x] code files in `grids`
- [ ] code files in `interpolation`
- [ ] [in progress] code files in `popsyn`
- [x] code files in `utils`
- [ ] code files in `visualization`

**Comment by sgossage** [comment]
@elizabethteng is working on [PR#499](https://github.com/POSYDON-code/POSYDON/pull/499) for unit tests in the `popsyn` module. Unit tests for the other modules have yet to be assigned.

We have decided however that further development on unit tests beyond `popsyn` are a low priority at the moment.

**Comment by astroJeff** [comment]
Converting from an issue to discussion. The question of how careful and complete we want our unit tests to be is a nuanced one
